File: src/types/studio.ts
export interface Studio {
  id: number;
  name: string;
  url: string | null;
  parent_id: number | null;
  details: string | null;
  rating: number | null;
  ignore_auto_tag: boolean;
  favorite: boolean;
  organized: boolean;
  created_at: string;
  updated_at: string;
}

export interface StudioCreate {
  name: string;
  url?: string;
  parent_id?: number;
  details?: string;
  rating?: number;
}

File: src-tauri/src/models/studio.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Studio {
    pub id: i64,
    pub name: String,
    pub url: Option<String>,
    pub parent_id: Option<i64>,
    pub details: Option<String>,
    pub rating: Option<i64>,
    pub ignore_auto_tag: bool,
    pub favorite: bool,
    pub organized: bool,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct StudioCreate {
    pub name: String,
    pub url: Option<String>,
    pub parent_id: Option<i64>,
    pub details: Option<String>,
    pub rating: Option<i64>,
}
